### PR TITLE
Enhancement: Add .editorconfig

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds an `.editorconfig`

💁‍♂️ For reference, see http://editorconfig.org.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.yml]
+indent_size = 2
```
Comment (localheinz) [inline .editorconfig]: I believe this configuration reflects what is *mostly* used within this project.

Comment (erayd) [reply .editorconfig]: While these settings may reflect current usage, they aren't necessarily desirable, and we should probably have a discussion around indentation particularly.

I'd rather like to see the entire project move to a 4-space indent, unless anyone else objects. What we have now is largely historical, and comes from a very old codebase with none of the original maintaners remaining.

Comment (localheinz) [reply .editorconfig]: Sounds good, @erayd!

Comment (bighappyface) [reply .editorconfig]: @localheinz shall we move to 4 space indention and move this forward?

Comment (localheinz) [reply .editorconfig]: @bighappyface 

Can do! Been busy today with tax preparation. Taking a look tomorrow!

Comment (localheinz) [reply .editorconfig]: @bighappyface 

Updated the indentation! 👍

**Pull request closed without merging**